### ISSUE TITLE
Circular dependency error

### ISSUE THREAD

**Issue by szydan** [jnoncode/decode-deps#9]
Hi 
Thanks for sharing this project 
I try to run it on a large code base and hit this error
Unfortunately I can not provide a small public example how to reproduce it, maybe the stack trace will help   

```
Module Dependency Graph Ready at http://localhost:5001
TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Object'
    |     property 'children' -> object with constructor 'Array'
    |     index 8 -> object with constructor 'Object'
    |     property 'children' -> object with constructor 'Array'
    --- index 2 closes the circle
    at JSON.stringify (<anonymous>)
    at stringify (/Users/szydan/.nvm/versions/node/v20.18.0/lib/node_modules/decode-deps/node_modules/express/lib/response.js:1160:12)
    at ServerResponse.json (/Users/szydan/.nvm/versions/node/v20.18.0/lib/node_modules/decode-deps/node_modules/express/lib/response.js:271:14)
    at file:///Users/szydan/.nvm/versions/node/v20.18.0/lib/node_modules/decode-deps/dist/route.js:18:9
    at Layer.handle [as handle_request] (/Users/szydan/.nvm/versions/node/v20.18.0/lib/node_modules/decode-deps/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/szydan/.nvm/versions/node/v20.18.0/lib/node_modules/decode-deps/node_modules/express/lib/router/route.js:149:13)
    at Route.dispatch (/Users/szydan/.nvm/versions/node/v20.18.0/lib/node_modules/decode-deps/node_modules/express/lib/router/route.js:119:3)
    at Layer.handle [as handle_request] (/Users/szydan/.nvm/versions/node/v20.18.0/lib/node_modules/decode-deps/node_modules/express/lib/router/layer.js:95:5)
    at /Users/szydan/.nvm/versions/node/v20.18.0/lib/node_modules/decode-deps/node_modules/express/lib/router/index.js:284:15
    at Function.process_params (/Users/szydan/.nvm/versions/node/v20.18.0/lib/node_modules/decode-deps/node_modules/express/lib/router/index.js:346:12)
```

**Comment by jnoncode** [comment]
Hello! Thank you for using this library.

There could be several causes for this error, but the most likely reason is **a circular dependency issue** in your project.

For example, this error may occur if `App1.tsx` exports a function that is imported into `App2.tsx`, and similarly, if `App2.tsx` exports a function that `App1.tsx` imports. This situation happens when two files reference each other. 

_(The following is an example of a situation that may cause this error)_
![example](https://github.com/user-attachments/assets/8514b5c8-6d02-4786-8555-70c365255e0e)

Currently, the library does not yet support handling circular dependencies, which is why this issue is occurring. I am actively working on solving this problem, and **an update should be available soon!**

Thank you very much for your interest and feedback. Your insights truly help me improve this project!

**Comment by jnoncode** [comment]
Hello!
This error has been resolved in version 1.1.0.
I recommend updating to the latest version and trying again.
If you still encounter the issue, please feel free to make new issue about it.
Thank you :)
